samples/csharp_dotnetcore/58.qnamaker-multiturn-sample/Dialog/QnAMakerMultiturnDialog.cs

```diff
@@ -0,0 +1,260 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.AI.QnA;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.BotBuilderSamples.Dialog
+{
+    /// <summary>
+    /// QnAMaker action builder class
+    /// </summary>
+    public class QnAMakerMultiturnDialog : ComponentDialog
+    {
+        // Dialog Options parameters	        // Dialog Options parameters
+        public const float DefaultThreshold = 0.3F;
+        public const int DefaultTopN = 3;
+        public const string DefaultNoAnswer = "No QnAMaker answers found."; 
+
+        // Card parameters	
+        public const string DefaultCardTitle = "Did you mean:";
+        public const string DefaultCardNoMatchText = "None of the above.";
+        public const string DefaultCardNoMatchResponse = "Thanks for the feedback.";
+
+
+        // Define value names for values tracked inside the dialogs.	
+        public const string QnAOptions = "qnaOptions";
+        public const string QnADialogResponseOptions = "qnaDialogResponseOptions";
+        private const string CurrentQuery = "currentQuery";
+        private const string QnAData = "qnaData";
+        private const string QnAContextData = "qnaContextData";
+        private const string PreviousQnAId = "prevQnAId";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="QnAMakerBaseDialog"/> class.
+        /// Dialog helper to generate dialogs.
+        /// </summary>
+        /// <param name="_services">Bot Services.</param>
+        public QnAMakerMultiturnDialog(IQnAMakerConfiguration services) : base(nameof(QnAMakerMultiturnDialog))
```

Comment:
Calling the parameter and the member variable services is odd now, can we call it configuration?
(Can we really not just use IConfiguration?)

Reply:
I have modified naming convention as configuration.